### PR TITLE
[Improvement][MOD-159] Add DOI message - Reviews

### PR DESCRIPTION
**Pull request by sheriefvt**
## Problem
Users don't know when they will have a DOI for their preprint.

## Solution
Add "DOI created after moderator approval." on pending pre-moderation preprints

image
image

## Ticket
https://openscience.atlassian.net/browse/MOD-159

### CODE REVIEW

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (aaxelb) [inline app/templates/preprints/provider/preprint-detail.hbs]: Delete this

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (aaxelb) [inline app/templates/preprints/provider/preprint-detail.hbs]: I think it'd be better to be more generic: display the message if `model.isPublished` is false. No matter the workflow, it will get a DOI when it transitions into a public state.

**Review by aaxelb — CHANGES_REQUESTED** [review]

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (laurenbarker) [reply app/templates/preprints/provider/preprint-detail.hbs]: We would need a different message for that because "after moderator approval" would be inaccurate for post-moderation or no moderation.

Could be "DOI created after preprint is made public".

Comment (aaxelb) [reply app/templates/preprints/provider/preprint-detail.hbs]: I'd be good with that.

Though for post-moderation, the only way `isPublished` would be false is if it's rejected, so it would require explicit approval. And with no moderation, this screen isn't even accessible.

Comment (laurenbarker) [reply app/templates/preprints/provider/preprint-detail.hbs]: Rejected, post-moderation preprints would have a DOI. You can still see the screen if you make your preprint if you make your node private though.

Comment (aaxelb) [reply app/templates/preprints/provider/preprint-detail.hbs]: Yeah, I assumed this would still be an `else if`, so just for preprints with no DOI. Once https://github.com/CenterForOpenScience/osf.io/pull/7837 is merged, we might be able to simplify and display this message whenever there's no DOI, which would be nice.

(If you make your node private, this page redirects to `page-not-found`. Or at least it should...)

Comment (laurenbarker) [reply app/templates/preprints/provider/preprint-detail.hbs]: It redirects to `page-not-found` for non-contributors.
screen shot 2017-10-25 at 3 21 03 pm

Comment (laurenbarker) [reply app/templates/preprints/provider/preprint-detail.hbs]: Whoops, this is my bad. I thought this change was on Preprints not Reviews. Disregard pretty much all of my comments.

Comment (sheriefvt) [reply app/templates/preprints/provider/preprint-detail.hbs]: @laurenbarker @aaxelb what changes do u suggest, and do you suggest change the wording

Comment (aaxelb) [reply app/templates/preprints/provider/preprint-detail.hbs]: ~~Change this line to `else if !model.isPublished` (or equivalent), and change wording to "DOI created after preprint is made public". Does that sound right @laurenbarker?~~
@sheriefvt Updated notes added to the ticket

**Review by aaxelb — APPROVED** [review]

**Pull request merged**
